src/Spice86.Core/Emulator/Devices/Input/Keyboard/Keyboard.cs

```diff
@@ -46,13 +50,43 @@ public class Keyboard : DefaultIOPortHandler {
     public override byte ReadByte(int port) {
         byte? scancode = LastKeyboardInput.ScanCode;
         if (scancode == null) {
-            return 0;
+            scancode = 0;
+        }
+        
+        return port switch {
+            KeyboardPorts.IOPort => scancode.Value,
+            // keyboard not locked, self-test completed.
+            KeyboardPorts.StatusRegister => 0x14,
+            _ => base.ReadByte(port)
+        };
+        
+    }
+
+    /// <inheritdoc />
+    public override void WriteByte(int port, byte value) {
+        switch (port) {
+            // the byte is interpreted as a data byte
+            case KeyboardPorts.IOPort:
+                if (Command == KeyboardCommand.SetOutputPort) {
+                    _machine.Memory.IsA20GateEnabled = (value & 2) > 0;
+                    Command = KeyboardCommand.None;
+                }
+                break;
+            case KeyboardPorts.CommandPort:
+                Command = value switch {
```

Comment:
Wouldnt it be clearer like that?
```
if(value == KeyboardValues.SetKeyboardCommand) {
  Command = KeyboardCommand.SetCommand;
}
```

Reply:
Done, and ports corrected by looking at the PCem code for the keyboard.